src/status_im/ui/screens/chat/views.cljs

```diff
@@ -265,39 +266,47 @@
     (utils/set-timeout #(re-frame/dispatch [:chat.ui/load-more-messages-for-current-chat])
                        (if platform/low-device? 700 200))))
 
-(defn messages-view [{:keys [chat bottom-space pan-responder space-keeper show-input?]}]
-  (let [{:keys [group-chat chat-id public?]} chat
-        messages @(re-frame/subscribe [:chats/chat-messages-stream chat-id])
-        current-public-key @(re-frame/subscribe [:multiaccount/public-key])]
+(defn messages-view [{:keys [chat bottom-space pan-responder space-keeper show-input? inverted-ui show-header show-footer]} messages render-fn]
+  (let [{:keys [group-chat chat-id public? community-id admins]} chat
+        current-public-key @(re-frame/subscribe [:multiaccount/public-key])
+        community @(re-frame/subscribe [:communities/community community-id])
+        group-admin? (get admins current-public-key)
+        community-admin? (when community (community :admin))
+        message-pin-enabled (and (not public?)
+                                 (or (not group-chat)
+                                     (and group-chat
+                                          (or group-admin?
+                                              community-admin?))))]
     ;;do not use anonymous functions for handlers
     [list/flat-list
      (merge
       pan-responder
       {:key-fn                       list-key-fn
        :ref                          list-ref
-       :header                       [list-header chat]
-       :footer                       [list-footer chat]
-       :data                         messages
-       :render-data                  {:group-chat         group-chat
-                                      :public?            public?
-                                      :current-public-key current-public-key
-                                      :space-keeper       space-keeper
-                                      :chat-id            chat-id
-                                      :show-input?        show-input?}
+       :header                       (when show-header [list-header chat])
+       :footer                       (when show-footer [list-footer chat])
+       :data                         (if inverted-ui (reverse @messages) @messages)
+       :render-data                  {:group-chat          group-chat
+                                      :public?             public?
+                                      :current-public-key  current-public-key
+                                      :space-keeper        space-keeper
+                                      :chat-id             chat-id
+                                      :show-input?         show-input?
+                                      :message-pin-enabled message-pin-enabled}
        :render-fn                    render-fn
        :on-viewable-items-changed    on-viewable-items-changed
```

Comment:
so for pin messages, the max number of items is 3 ? and we don't need paging for pinned messages right ? in that case its better to do not share messages-view and just add `get-render-data` function and share it?

Reply:
Exactly, and makes sense as that should share most of the subscriptions. Moved `render-data` to a separate function now